### PR TITLE
Fixed the routing for filters

### PR DESCRIPTION
**Pull request by hemny-singh**

### CODE REVIEW

File: squealy/squealy-web/src/Containers/AuthoringInterfaceContainer.js [hunk 1]
```diff
@@ -78,30 +80,44 @@ export default class AuthoringInterfaceContainer extends Component {
   }
 
   // Updates the selected chart index and updates the selected chart name in the URL
-  onWidgetDeleted = (index, selectedWidgetStateKey, widgetStateKeyData, callback) => {
-    const selectedIndex = this.state[selectedWidgetStateKey],
+  onWidgetDeleted = (index, selectedIndexStateKey, widgetStateKeyData, callback) => {
```
Comment (devashishsharma2302) [inline squealy/squealy-web/src/Containers/AuthoringInterfaceContainer.js]: Ambiguous naming. can be replaced by selectedWidgetIndex or something else.

File: squealy/squealy-web/src/Containers/AuthoringInterfaceContainer.js [hunk 2]
```diff
@@ -78,30 +80,44 @@ export default class AuthoringInterfaceContainer extends Component {
...
+    const selectedIndex = this.state[selectedIndexStateKey] || index,
       currWidgetData = this.state[widgetStateKeyData]
+    let chartMode = false, nonSelectedStatIndex = ''
+
+    if (selectedIndexStateKey === 'selectedChartIndex') {
+      nonSelectedStatIndex = 'selectedFilterIndex'
```
Comment (devashishsharma2302) [inline squealy/squealy-web/src/Containers/AuthoringInterfaceContainer.js]: Spelling mistake, also it should be renamed to something like deselectedWidgetIndex.

File: squealy/squealy-web/src/Components/ParamDefinitionModal.js [hunk 3]
```diff
@@ -153,7 +173,8 @@ export default class ParamDefinitionModal extends Component {
         errorName: false,
         errorTestValue: false,
         errorDefaultValue: false,
-        editArrayIndex: index
+        editArrayIndex: index,
+        selectedDropdownAPI: selectedDropdownAPI
```
Comment (hemny-singh) [inline squealy/squealy-web/src/Components/ParamDefinitionModal.js]: It's more specific to dropdown filter. Should I still change the name?

File: squealy/parameters.py [hunk 4]
```diff
@@ -121,4 +121,25 @@ def to_internal(self, value):
             else :
                 return float(value)
         except ValueError:
-            raise NumberParseException("Cannot parse to int or float"+ value)
\ No newline at end of file
+            raise NumberParseException("Cannot parse to int or float"+ value)
+
+
+class Dropdown(Parameter):
```
Comment (swapnil95) [inline squealy/parameters.py]: Inherit from String class as all the methods are same as that of String class

**Pull request merged**